pkg/update/update.go

```diff
@@ -283,3 +292,17 @@ func makeWaitUntilAllPodsReadyFuncInAllClusters(
 		return backoff.Retry(f, b)
 	}
 }
+
+// TODO there are ALOT more reason codes in k8sErrors, should we test them all?
```

Comment:
Do the k8s errors not provide enough context/information on their own?

Reply:
sorta.  The way the code is written we need to act differently depending on the error rather than just assuming that the pod is not found.  For instance, if we get an authz error on an API call, we should fail, and not assume that the pod is not found.  There are ALOT of different types of errors, and the logic needs improving as well in our code.